compiler/semantic/op.go

```diff
@@ -915,70 +813,55 @@ func (a *analyzer) singletonAgg(agg ast.Assignment, seq dag.Seq) dag.Seq {
 	)
 }
 
-func (a *analyzer) semDecls(decls []ast.Decl) ([]dag.Def, []*dag.Func, error) {
+func (a *analyzer) semDecls(decls []ast.Decl) ([]dag.Def, []*dag.Func) {
 	var consts []dag.Def
 	var fnDecls []*ast.FuncDecl
 	for _, d := range decls {
 		switch d := d.(type) {
 		case *ast.ConstDecl:
-			c, err := a.semConstDecl(d)
-			if err != nil {
-				return nil, nil, err
-			}
-			consts = append(consts, c)
+			consts = append(consts, a.semConstDecl(d))
 		case *ast.FuncDecl:
 			fnDecls = append(fnDecls, d)
 		case *ast.OpDecl:
-			if err := a.semOpDecl(d); err != nil {
-				return nil, nil, err
-			}
+			a.semOpDecl(d)
 		case *ast.TypeDecl:
-			c, err := a.semTypeDecl(d)
-			if err != nil {
-				return nil, nil, err
-			}
-			consts = append(consts, c)
+			consts = append(consts, a.semTypeDecl(d))
 		default:
-			return nil, nil, fmt.Errorf("invalid declaration type %T", d)
+			panic(fmt.Errorf("invalid declaration type %T", d))
 		}
 	}
-	funcs, err := a.semFuncDecls(fnDecls)
-	if err != nil {
-		return nil, nil, err
-	}
-	return consts, funcs, nil
+	funcs := a.semFuncDecls(fnDecls)
+	return consts, funcs
 }
 
-func (a *analyzer) semConstDecl(c *ast.ConstDecl) (dag.Def, error) {
-	e, err := a.semExpr(c.Expr)
-	if err != nil {
-		return dag.Def{}, err
-	}
+func (a *analyzer) semConstDecl(c *ast.ConstDecl) dag.Def {
+	e := a.semExpr(c.Expr)
 	if err := a.scope.DefineConst(a.zctx, c.Name, e); err != nil {
-		return dag.Def{}, err
+		a.error(c, err)
 	}
 	return dag.Def{
 		Name: c.Name,
 		Expr: e,
-	}, nil
+	}
 }
 
-func (a *analyzer) semTypeDecl(d *ast.TypeDecl) (dag.Def, error) {
+func (a *analyzer) semTypeDecl(d *ast.TypeDecl) dag.Def {
 	typ, err := a.semType(d.Type)
 	if err != nil {
-		return dag.Def{}, err
+		a.error(d.Type, err)
+		typ = "error(\"string\")"
```

Comment:
Nit: `null` feels more appropriate to me.
```suggestion
		typ = "null"
```